py/sentry_relay/consts.py

```diff
@@ -1,25 +1,92 @@
+from enum import IntEnum
+
+from sentry_relay._lowlevel import lib
+from sentry_relay.utils import decode_str, encode_str
+
+
 __all__ = ["SPAN_STATUS_CODE_TO_NAME", "SPAN_STATUS_NAME_TO_CODE"]
 
 
-SPAN_STATUS_CODE_TO_NAME = {
-    0: "ok",
-    1: "cancelled",
-    2: "unknown_error",
-    3: "invalid_argument",
-    4: "deadline_exceeded",
-    5: "not_found",
-    6: "already_exists",
-    7: "permission_denied",
-    8: "resource_exhausted",
-    9: "failed_precondition",
-    10: "aborted",
-    11: "out_of_range",
-    12: "unimplemented",
-    13: "internal_error",
-    14: "unavailable",
-    15: "data_loss",
-    16: "unauthenticated",
-}
-
-SPAN_STATUS_NAME_TO_CODE = dict((v, k) for k, v in SPAN_STATUS_CODE_TO_NAME.items())
-SPAN_STATUS_NAME_TO_CODE["unknown"] = SPAN_STATUS_NAME_TO_CODE["unknown_error"]
+class BaseDataCategory(object):
+    @classmethod
+    def parse(cls, name):
+        """
+        Parses a `DataCategory` from its API name.
+        """
+        category = cls(lib.relay_data_category_parse(encode_str(name or "")))
+        if category == cls.UNKNOWN:
+            return None  # Unknown is a Rust-only value, replace with None
+        return category
+
+    @classmethod
+    def from_event_type(cls, event_type):
+        """
+        Parses a `DataCategory` from an event type.
+        """
+        s = encode_str(event_type or "")
+        return cls(lib.relay_data_category_from_event_type(s))
+
+    @classmethod
+    def event_categories(cls):
+        """
+        Returns categories that count as events, including transactions.
+        """
+        return [cls.DEFAULT, cls.ERROR, cls.TRANSACTION, cls.SECURITY]
+
+    @classmethod
+    def error_categories(cls):
+        """
+        Returns categories that count as traditional error tracking events.
+        """
+        return [cls.DEFAULT, cls.ERROR, cls.SECURITY]
+
+    def api_name(self):
+        """
+        Returns the API name of the given `DataCategory`.
+        """
+        return decode_str(lib.relay_data_category_name(self.value), free=True)
+
+
+def _make_data_categories():
+    prefix = "RELAY_DATA_CATEGORY_"
+    categories = {}
+
+    for attr in dir(lib):
+        if not attr.startswith(prefix):
+            continue
+
+        category_name = attr[len(prefix) :]
+        categories[category_name] = getattr(lib, attr)
+
+    data_categories = IntEnum(
+        "DataCategory", categories, type=BaseDataCategory, module=__name__
+    )
+    globals()[data_categories.__name__] = data_categories
+    __all__.append(data_categories.__name__)
+
+
+_make_data_categories()
+
+
+SPAN_STATUS_CODE_TO_NAME = {}
+SPAN_STATUS_NAME_TO_CODE = {}
+
+
+def _make_span_statuses():
+    prefix = "RELAY_SPAN_STATUS_"
+
+    for attr in dir(lib):
+        if not attr.startswith(prefix):
+            continue
+
+        status_name = attr[len(prefix) :].lower()
```

Comment:
I believe that this is actually safe - per convention the constants must always match the default serialization. That said, if we ever diverge from this in the future (which we should not!) we can call into an API for this.

Reply:
Yes this seems safe.